bricepollock/canyoneer
# [ISSUE-19] enable manually updating from the server

### Description
Updates from server every week by checking the index file and pull down updates for any individual canyons which have changed.

Failure cases:
Does not write to system a new index file unless the full updates has succeeded. This means upon any failure, it appears to the system no update has occurred on next launch. This ensures that we don't write bad-state to the filesystem.

Note: There a number of compiler flags so then when in DEBUG or TEST we have certain behaviors that help us write tests or debug while building from Xcode. The hope is that these things will also be disabled or assert for production build. We will have to test via TestFlight to confirm for sure

### Test Plan
* Manually update from bundled canyons to server
* Ensure that favorites are migrated as expected across both App Updates and then after updating from download
* Trigger manual flushing of background tasks via iPhone device to determine that background update works:
```
e -l objc -- (void)[[BGTaskScheduler sharedScheduler] _simulateLaunchForTaskWithIdentifier:@"backcountry_nomad_canyoneer_index_update"]
```

* Use push notifications to debug that background updates are working as expected
